desc/grid.py

```diff
@@ -447,17 +455,33 @@ def _create_nodes(
             if self.sym:
                 t += t[1] / 2
             dt = 2 * np.pi / t.size * np.ones_like(t)
+
         else:
             t = np.atleast_1d(theta)
             dt = np.zeros_like(t)
             if t.size > 1:
-                dt[0] = (t[0] + t[1]) / 2
+                # in case t[0] != 0, we add it evenly between dt[0] and dt[-1]
+                dt[0] = t[1] - t[0] / 2
                 dt[1:-1] = (t[2:] - t[:-2]) / 2
-                dt[-1] = 2 * np.pi - (t[-2] + t[-1]) / 2
+                dt[-1] = 2 * np.pi - t[-1] + t[0] / 2
+                # maybe below 2 lines would be better than above 3 lines
+                # dt = np.diff(t, append=2 * np.pi + t[0] / 2)
+                # dt[0] += t[0] / 2
+                dt *= (t.size - endpoint) / t.size
```

Comment:
wouldn't this give `dt=0` if there's only 1 theta coordinate and `endpoint=True`?

Reply:
wait nvm i see this is inside the if block

Reply:
Yes, it's in the if block. Line 470 is pretty opaque. Basically, lines 463-466 will assign spacing in a way that already scales the weights down for repeated nodes similar to `_scale_weights`. So if there is an extra node at 2pi, I upscale the weights so that `_scale_weights` can do it's job.